tests/ConsumerPactTests.cs

```diff
@@ -1,61 +1,68 @@
         {
-            _mockProviderService = fixture.MockProviderService;
-            _mockProviderService.ClearInteractions(); //NOTE: Clears any previously registered interactions before the test is run
-            _mockProviderServiceBaseUri = fixture.MockProviderServiceBaseUri;
+
+            products = new List<object>()
+            {
+                new { id = "27", name = "burger", type = "food" }
+            };
+
+            var Config = new PactConfig
+            {
+                PactDir = Path.Join("..", "..", "..", "..", "pacts"),
+                Outputters = new[] { new XunitOutput(output) },
+                LogLevel = PactLogLevel.Debug
+            };
+
+            pact = Pact.V3("pactflow-example-consumer-dotnet", "pactflow-example-provider-dotnet", Config).WithHttpInteractions(port);
         }
 
         [Fact]
         public async void RetrieveProducts()
         {
             // Arrange
-            _mockProviderService.Given("products exist")
-                                .UponReceiving("A request to get products")
-                                .With(new ProviderServiceRequest
-                                {
-                                    Method = HttpVerb.Get,
-                                    Path = "/products",
-                                })
-                                .WillRespondWith(new ProviderServiceResponse {
-                                    Status = 200,
-                                    Headers = new Dictionary<string, object>
-                                    {
-                                        { "Content-Type", "application/json; charset=utf-8" }
-                                    },
-                                    Body = new MinTypeMatcher(new
-                                    {
-                                        id = "27",
-                                        name = "burger",
-                                        type = "food"
-                                    }, 1)
-                                });
-
-            // Act
-            var consumer = new ProductClient();
-            List<Product> result = await consumer.GetProducts(_mockProviderServiceBaseUri);
-
-            // Assert
-            result.Should().NotBeNull();
-            result.Should().HaveCount(1);
-            result[0].id.Should().Equals("27");
-            result[0].name.Should().Equals("burger");
-            result[0].type.Should().Equals("food");
+            pact.UponReceiving("A request to get products")
+                        .Given("products exist")
+                        .WithRequest(HttpMethod.Get, "/products")
+                    .WillRespond()
+                    .WithStatus(HttpStatusCode.OK)
+                    .WithHeader("Content-Type", "application/json; charset=utf-8")
+                    .WithJsonBody(new TypeMatcher(products));
+
+            await pact.VerifyAsync(async ctx =>

```

Comment:
We wrap our system under test `ProductClient().GetProducts(...)` inside a `VerifyAsync` block.

This starts our `Pact.V3(...)` mock provider, and can give us access to a dynamically created URL/Port if required.

We then perform our assertions on our result from our consumer method `GetProducts(...)` rather than on the expected Providers response. This is the traditional unit testing style, and ensures you are testing consumer behaviour rather than providers implementation detail